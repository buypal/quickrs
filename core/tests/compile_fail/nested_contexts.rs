use rquickjs::{Context, Function, Runtime, String};

fn main() {
    let rt = Runtime::new().unwrap();
    let ctx_1 = Context::full(&rt).unwrap();
    let ctx_2 = Context::full(&rt).unwrap();
    ctx_1.with(|ctx_1| {
        let val: String = ctx_1.eval("'foo'").unwrap();
        ctx_2.with(|ctx_2| {
            let f: Function = ctx_2.eval("x => x + 'b'").unwrap();
            f.call::<_, ()>(val).unwrap();
        })
    })
}
