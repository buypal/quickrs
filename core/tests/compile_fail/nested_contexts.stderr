error[E0495]: cannot infer an appropriate lifetime for lifetime parameter `'js` due to conflicting requirements
  --> $DIR/nested_contexts.rs:8:33
   |
8  |         let val: String = ctx_1.eval("'foo'").unwrap();
   |                                 ^^^^
   |
note: first, the lifetime cannot outlive the anonymous lifetime #2 defined on the body at 9:20...
  --> $DIR/nested_contexts.rs:9:20
   |
9  |           ctx_2.with(|ctx_2| {
   |  ____________________^
10 | |             let f: Function = ctx_2.eval("x => x + 'b'").unwrap();
11 | |             f.call::<_, ()>(val).unwrap();
12 | |         })
   | |_________^
note: ...but the lifetime must also be valid for the anonymous lifetime #2 defined on the body at 7:16...
  --> $DIR/nested_contexts.rs:7:16
   |
7  |       ctx_1.with(|ctx_1| {
   |  ________________^
8  | |         let val: String = ctx_1.eval("'foo'").unwrap();
9  | |         ctx_2.with(|ctx_2| {
10 | |             let f: Function = ctx_2.eval("x => x + 'b'").unwrap();
11 | |             f.call::<_, ()>(val).unwrap();
12 | |         })
13 | |     })
   | |_____^
note: ...so that the types are compatible
  --> $DIR/nested_contexts.rs:10:37
   |
10 |             let f: Function = ctx_2.eval("x => x + 'b'").unwrap();
   |                                     ^^^^
   = note: expected `Ctx<'_>`
              found `Ctx<'_>`
